### ISSUE TITLE
Focus not visible on block types in High Contrast modes

### ISSUE THREAD

**Issue by karlgroves** [WordPress/gutenberg#15327]
Focus not visible on block types in High Contrast modes

-   **Severity:** Medium
-   **Affected Populations:**
    -   Low-Vision
    -   Motor Impaired
-   **Platform(s):**
    -   Windows - High Contrast
-   **Components affected:**
    -   Add Block

#### Issue description

When users of Windows High Contrast mode navigate between potential
block types, an outline is added to the currently-focussed block.
However this outline is pulled inwards with an `outline-offset`
property, which sets it directly over the border. This makes it
difficult to tell which element is focussed in browsers which support
that property (currently all except IE).

All interactive elements should have a visible and obvious focus
indicator even for high contrast users, so that sighted keyboard users
have a clear indication of which element has focus.

##### Issue Code

```
    .editor-block-types-list__item:not(:disabled).is-active, .editor-block-types-list__item:not(:disabled):active, .editor-block-types-list__item:not(:disabled):focus {


        outline: 0;


        ...


        outline: 2px solid transparent;


        outline-offset: -2px;


    }
```

#### Remediation Guidance

Remove the `outline-offset` rule and allow the outline to sit outside
the border.

##### Recommended Code

```
    .editor-block-types-list__item:not(:disabled).is-active, .editor-block-types-list__item:not(:disabled):active, .editor-block-types-list__item:not(:disabled):focus {


        ...


        outline: 2px solid transparent;


    }

```

#### Relevant standards

-   [2.4.7 Focus Visible (Level
    AA)](https://www.w3.org/TR/WCAG20/#navigation-mechanisms-focus-visible)

**Note:** This issue may be a duplicate with other existing accessibility-related bugs in this project. This issue comes from the Gutenberg accessibility audit, performed by [Tenon](https://www.tenon.io) and funded by [WP Campus](https://wpcampus.org/). This issue is GUT-17 in Tenon's report

**Comment by melchoyce** [comment]
Screenshot from full report:

![image](https://user-images.githubusercontent.com/2846578/57184701-18a15a00-6e74-11e9-877c-1792e3640c0b.png)

**Comment by afercia** [comment]
Direct comparison: 
- on the left: IE11 – focus transparent outline is visible (no support for outline-offset)
- on the right: Edge – focus transparent outline is not visible (does support outline-offset)

![Screenshot (221)](https://user-images.githubusercontent.com/1682452/57389094-c3d64a00-71b9-11e9-9478-5110d00b0967.png)

As far as I remember, the `outline-offset` was implemented because at that time it looked better on _some_ buttons. As it doesn't work correctly for all elements, simplest option is to just remove it.

Worth noting there are other elements where the negative offset makes the transparent outline not visible, for example the Block toolbar buttons. See screenshot below, where the "Bold" button is focused (Edge):

![Screenshot (223)](https://user-images.githubusercontent.com/1682452/57389039-ad2ff300-71b9-11e9-8646-199dcdb558ff.png)

**Comment by tellthemachines** [comment]
Closing this issue as it was fixed by #16554.
